src/core/simulation_stages.jl

```diff
@@ -3,20 +3,22 @@ mutable struct StageInternal
     number::Int
     executions::Int
     execution_count::Int
-    synchronized_executions::Dict{Int, Int} # Number of executions per upper level stage step
+    end_of_interval_step::Int
+    # This line keeps track of the executions of a stage relative to other stages.
+    # This might be needed in the future to run multiple stages. For now it is disabled
+    #synchronized_executions::Dict{Int, Int} # Number of executions per upper level stage step
     psi_container::Union{Nothing, PSIContainer}
-    cache_dict::Dict{Type{<:AbstractCache}, AbstractCache}
-    # Can probably be eliminated and use getter functions from
-    # Simulation object. Need to determine if its always available in the stage update steps.
+    caches::Set{<:CacheKey{<:AbstractCache, <:PSY.Device}}
```

Comment:
Can you add a comment on `caches`, particularly explaining why it is a Set?

Reply:
I made it a set because I don't need any order and I need to make sure that there is only one of each.